hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/AvroConversionUtils.scala

```diff
@@ -98,13 +98,10 @@ object AvroConversionUtils {
    */
   def createDataFrame(rdd: RDD[GenericRecord], schemaStr: String, ss: SparkSession): Dataset[Row] = {
     ss.createDataFrame(rdd.mapPartitions { records =>
-      if (records.isEmpty) Iterator.empty
-      else {
-        val schema = new Schema.Parser().parse(schemaStr)
-        val dataType = convertAvroSchemaToStructType(schema)
-        val converter = createConverterToRow(schema, dataType)
-        records.map { r => converter(r) }
-      }
+      val schema = new Schema.Parser().parse(schemaStr)
```

Comment:
will it yields an empty data frame here?

Reply:
if (records.isEmpty) Iterator.empty
So wouldn't calling records.map still be empty?
No one I've asked so far knows why we did this, but I am also hesitant to make this change as well

Reply:
yeah, the gains for `records.isEmpty` is little, do we have a test for empty records here?

Reply:
Let's test all the relevant code path involving `records.isEmpty` with empty Spark partitions on multiple Spark versions to make sure the removal of `#isEmpty` is safe.